src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -144,6 +160,515 @@ TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
+      .WillOnce(Return(mock_app_nonexistent));
+
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE;
+  (*msg)[am::strings::msg_params][am::strings::info] = "info1";
+
+  Event event(hmi_apis::FunctionID::UI_SetAppIcon);
+  event.set_smart_object(*msg);
+
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).Times(0);
+  req_vr->on_event(event);
+}
+
+TEST_F(SetAppIconRequestTest, Run_AllCorrectData_SendHMIRequest) {
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg_vr);
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillRepeatedly(Return(mock_app));
+
+  MockApplicationManagerSettings mock_app_mngr_settings;
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillRepeatedly(ReturnRef(mock_app_mngr_settings));
+
+  const std::string app_storage_folder = "./storage";
+  EXPECT_CALL(mock_app_mngr_settings, app_storage_folder())
+      .WillRepeatedly(ReturnRef(app_storage_folder));
+
+  const std::string app_folder = "Test_app";
+  EXPECT_CALL((*mock_app), folder_name()).WillRepeatedly(Return(app_folder));
+
+  MockProtocolHandler mock_app_protocol_handler;
+  EXPECT_CALL(app_mngr_, protocol_handler())
+      .WillRepeatedly(ReturnRef(mock_app_protocol_handler));
+
+  MockProtocolHandlerSettings mock_protocol_handler_settings;
+  EXPECT_CALL(mock_app_protocol_handler, get_settings())
+      .WillRepeatedly(ReturnRef(mock_protocol_handler_settings));
+
+  const protocol_handler::MajorProtocolVersion mock_app_mngr_protocol_version =
+      protocol_handler::PROTOCOL_VERSION_4;
+  EXPECT_CALL(mock_protocol_handler_settings, max_supported_protocol_version())
+      .WillRepeatedly(Return(mock_app_mngr_protocol_version));
+
+  const std::string icon_folder = app_storage_folder + "/" + app_folder;
+  EXPECT_CALL(mock_app_mngr_settings, app_icons_folder())
+      .WillRepeatedly(ReturnRef(icon_folder));
+
+  const uint32_t storage_max_size = 10u;
+  EXPECT_CALL(mock_app_mngr_settings, app_icons_folder_max_size())
+      .WillRepeatedly(ReturnRef(storage_max_size));
+
+  const std::string app_policy_id = "test_icon";
+  EXPECT_CALL((*mock_app), policy_app_id())
+      .WillRepeatedly(Return(app_policy_id));
+
+  const uint32_t app_id = 152u;
+  EXPECT_CALL((*mock_app), app_id()).WillRepeatedly(Return(app_id));
+
+  MessageSharedPtr result;
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  const std::string directory = app_storage_folder + "/" + app_folder;
+  const std::string full_file_path = directory + "/" + kSyncFileName;
+  file_system::CreateDirectoryRecursively(directory);
+  file_system::CreateFile(full_file_path);
+  req->Run();
+  file_system::RemoveDirectory(app_storage_folder);
+
+  const std::string expect_sync_file_name =
```

Comment:
@BSolonenko psopose to rename all your `expect_*` variables to `expected_*` as its more correct in this context

Reply:
@AKalinich-Luxoft @AByzhynar  Fixed in https://github.com/smartdevicelink/sdl_core/pull/1974/commits/beb0fd60bd310bdbc5a04821177a9005cb5c0fd9